tasks/main.yml

```diff
@@ -127,6 +135,19 @@
       set_fact:
         ansible_python_interpreter: "{{ os_images_venv }}/bin/python"
 
+    - name: Ensure existing cloud tenant kernel does not exist
```

Comment:
In kayobe, ansible/roles/ipa-images/tasks/main.yml, I use image checksums to improve the conditional upload. Could be worth trying here.

Reply:
If it is acceptable to have a requirement on ansible 2.5 we could use `os_image_facts` without the `image` parameter to get the checksums in a single query. Otherwise, we have to make a query for every image - I'm not sure if this would end up being slower than just always updating the image?

Reply:
Good question. Right now we don't limit the version of galaxy dependencies for kayobe etc. For the stable/pike branch, we can't use ansible 2.5. I guess the fix is to be more proscriptive in kayobe's galaxy requirements. Not sure whether this should reference a tag, or commit ID, or what.

Reply:
I suppose when forcing rebuilds we can fairly safely delete the image. What doesn't work currently though is if you build an image that is already in glance - the new one won't be uploaded unless you delete the old one.

Another thing to consider is whether we need to do some sort of life cycle management for the images, rather than just deleting them. If anything is referencing the image then it could be broken by a simple delete. I guess we could say that that is a separate problem, and not compatible with force-rebuild.

Reply:
Here is a partial ansible 2.4 :

https://github.com/stackhpc/ansible-role-os-images/commit/e78499a812e9a6370b43e13d33cc1956ebb35987

Reply:
Was trying to keep the lists aligned for simplicity...

Reply:
Nice work with the checksums, let's get this one merged first.